sstables/checksummed_data_source.cc

```diff
@@ -88,9 +102,18 @@ class checksummed_file_data_source_impl : public data_source_impl {
             if (expected_checksum != actual_checksum) {
                 throw sstables::malformed_sstable_exception(format("Checksummed chunk of size {} at file offset {} failed checksum: expected={}, actual={}", buf.size(), _underlying_pos, expected_checksum, actual_checksum));
             }
+
+            if (_expected_digest) {
+                _actual_digest = checksum_combine_or_feed<ChecksumType>(_actual_digest, actual_checksum, buf.begin(), buf.size());
+            }
+
             buf.trim_front(_pos & (chunk_size - 1));
             _pos += buf.size();
             _underlying_pos += chunk_size;
+
+            if (_pos == _file_len && _expected_digest && *_expected_digest != _actual_digest) {
+                throw malformed_sstable_exception(format("Digest mismatch: expected={}, actual={}", *_expected_digest, _actual_digest));
```

Comment:
Qualify format() calls with the namespace to avoid ambiguity, here fmt::format() is indicated since you want std::string.

Reply:
Actually `malformed_sstable_exception` takes an `sstring`. I'll add a `seastar::` prefix.